crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -752,14 +752,9 @@ impl CumulocityConverter {
         for child_id in local_child_devices {
             // update the children cache with the operations supported
             let ops = Operations::try_new(path_to_child_devices.join(&child_id))?;
-            let child_topic_id = EntityTopicId::default_child_device(&child_id).unwrap();
-            let child_external_id =
-                Self::map_to_c8y_external_id(&child_topic_id, &self.device_name.as_str().into());
-
-            self.children
-                .insert(child_external_id.as_ref().into(), ops.clone());
+            self.children.insert(child_id.clone(), ops.clone());
             let ops_msg = ops.create_smartrest_ops_message()?;
-            let topic_str = format!("{SMARTREST_PUBLISH_TOPIC}/{}", child_external_id.as_ref());
+            let topic_str = format!("{SMARTREST_PUBLISH_TOPIC}/{}", child_id);
```

Comment:
issue: `child_id` which comes from `local_child_devices`, which in turn is derived from names  of filesystem directories, can also be invalid MQTT topics. To fix this, we need to try to parse it as `EntityTopicId` and if it's incorrect, return `Result::Err` from the function, in effect aborting on first invalid input. This can be improved later, by e.g. successfully processing all valid children and reporting any invalid children.

Reply:
note: there is a `C8yTopic` type that we should ideally be using for usecases like this, but currently it's a little unwieldy because it takes the entire `EntityMetadata`, so for now this manual `format!` is fine, as modifying `C8yTopic` is outside the scope of this PR.